template-parts/post-list-filters.php

```diff
@@ -0,0 +1,149 @@
+				$filter_count = 0;
+
+				// Search term.
+				if ( ! empty( $query_var_search_term ) ) {
+					$filter_count++;
+				}
+
+				// Date interval.
+				if ( ! empty( $query_var_date_from ) && ! empty( $query_var_date_to ) ) {
+					$filter_count++;
+				}
+
+				// Categories.
+				$filter_count += count( $query_var_categories );
+
+				printf( esc_html__( 'Show filters', 'shiro' ) );
+
+				if ( $filter_count > 0 ) {
+					echo ' <em>';
+					/* translators: 1. how many filters were applied */
+					printf( esc_html__( '(%s applied)', 'shiro' ), esc_html( $filter_count ) );
+					echo '</em>';
+				}
+			} else {
+				printf( esc_html__( 'Show filters', 'shiro' ) );
+			}
+			?>
+			</span>
+		</button>
+
+	</div>
+
+	<form method="get" class="post-list-filter__form">
+
+		<?php wp_nonce_field( 'post_list_filters', 'post_list_filters_nonce' ); ?>
+
+		<div class="post-list-filter__container mw-980">
+
+			<div class="filter-by-text">
+				<h5>
+					<?php printf( esc_html__( 'Filter by text', 'shiro' ) ); ?>
+					<?php if ( ! empty( $query_var_search_term ) ) : ?>
+						&nbsp;<em><u>(<?php printf( esc_html__( 'applied', 'shiro' ) ); ?>)</u></em>
+					<?php endif; ?>
+				</h5>
+				<div class="search-text-input-button">
+					<input type="text" name="s" value="<?php echo esc_attr( $query_var_search_term ); ?>">
+				</div>
+			</div>
+
+			<div class="filter-by-date">
+				<h5>
+					<?php printf( esc_html__( 'Filter by date', 'shiro' ) ); ?>
+					<?php if ( ! empty( $query_var_date_from ) && ! empty( $query_var_date_to ) ) : ?>
+						&nbsp;<em><u>(<?php printf( esc_html__( 'applied', 'shiro' ) ); ?>)</u></em>
+					<?php endif; ?>
+				</h5>
+				<div class="filter-date-inputs-container">
+					<input type="date" name="date_from" placeholder="From" value="<?php echo esc_attr( $query_var_date_from ); ?>">
+					<input type="date" name="date_to" placeholder="To" value="<?php echo esc_attr( $query_var_date_to ); ?>">
+					<button type="button" class="action-button action-button--clear" id="button-reset-date-filters"><?php printf( esc_html__( 'Reset', 'shiro' ) ); ?></button>
+				</div>
+			</div>
+
+			<div class="filter-by-category">
+				<?php
+				$categories = get_categories();
+
+				foreach ( $categories as $category ) {
+					$category_display = ( $category->parent == 0 )
+						? $category->name
+						: get_category_parents( $category->parent, false, ' > ', false ) . $category->name;
+
+					$categories_array[ $category->slug ] = $category_display;
+				}
+				asort( $categories_array );
+				?>
+
+				<h5>
+					<?php printf( esc_html__( 'Filter by category', 'shiro' ) ); ?>
+					<?php if ( count( $query_var_categories ) > 0 ) : ?>
+						&nbsp;<em><u>(<?php echo count( $query_var_categories ) . ' ' . esc_html__( 'applied', 'shiro' ); ?>)</u></em>
+					<?php endif; ?>
+				</h5>
+
+				<ul class='category-container'>
+					<?php foreach ( $categories_array as $category->slug => $category_display ) : ?>
```

Comment:
This foreach statement looks weird, does it do what you'd expect? I'd expect the key to have to be a simple variable like `$category_slug`

Reply:
Understood  `_n`, nice trick, thanks!

About the loop, I got the problem - it was _strangely_ working because I was rewriting the object property defined above. Fixed it!